vyper/ast/parse.py

```diff
@@ -401,7 +395,21 @@ def visit_Constant(self, node):
         if node.value is None or isinstance(node.value, bool):
             node.ast_type = "NameConstant"
         elif isinstance(node.value, str):
-            node.ast_type = "Str"
+            if (node.lineno, node.col_offset) in self._native_hex_literal_locations:
+                if len(node.value) % 2 != 0:
+                    raise SyntaxException(
+                        "Native hex string must have an even number of characters",
+                        self._source_code,
+                        node.lineno,
+                        node.col_offset,
+                    )
+
+                byte_val = bytes.fromhex(node.value)
+
+                node.ast_type = "Bytes"
```

Comment:
i am not a huge fan of fusing it into the Bytes type, since we lose information in the AST. let's make a new VyperNode AST type, maybe `XBytes` or `HexBytes`?